src/shape.rs

```diff
@@ -93,6 +93,25 @@ impl Shape {
         }
     }
 
+    /// Obtains dimensions as an array.
+    ///
+    /// # Returns
+    ///
+    /// * `Ok([usize; N])` - Array representation of dimensions.
+    /// * `Err(Error)` - The shape is not N-dimensional.
+    pub fn as_array<const N: usize>(&self) -> Result<[usize; N]> {
+        if N == self.num_dimensions {
+            let mut v = MaybeUninit::<[usize; N]>::uninit();
+            unsafe { &mut *v.as_mut_ptr() }.copy_from_slice(&self.dimensions[..N]);
+            Ok(unsafe { v.assume_init() })
```

Comment:
https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#method.as_mut_ptr
> Reading from this pointer or turning it into a reference is undefined behavior unless the MaybeUninit<T> is initialized.

However, the result of `as_mut_ptr()` is turned into a reference before initialization in this code.
The current code may work, but why not use the implementation of `uninit_array()`, `write_slice()`, and `array_assume_init()` provided by Nightly Rust?

```suggestion
            // let mut v: [MaybeUninit<usize>; N] = MaybeUninit::uninit_array();
            let mut v = unsafe { MaybeUninit::<[MaybeUninit<usize>; N]>::uninit().assume_init() };
            // MaybeUninit::write_slice(&mut v, &self.dimensions[..N]);
            let uninit_dimensions: &[MaybeUninit<T>] = unsafe { std::mem::transmute(self.dimensions[..N]) };
            v.copy_from_slice(uninit_dimensions);
            // Ok(unsafe { MaybeUninit::array_assume_init(v) })
            Ok(unsafe { (&v as *const _ as *const [usize; N]).read() })
```

Reply:
because it is nightly.

Reply:
I think this UB would become serious sometimes and needs to be avoided if possible.

Reply:
What does UB mean?

Reply:
ahh, undefined behaviour

Reply:
It seems the current version of the language could not define types related to const parameters. This implementation needs to be modified.

Reply:
fixed in 928d375, now we have separate definitions for all `as_array[0-8]`.

Reply:
I guessed this is why ndarray defines separate ArrayN instead of using generics...